components/select_team/select_team.jsx

```diff
@@ -57,11 +60,24 @@ export default class SelectTeam extends React.Component {
         this.state = {
             loadingTeamId: '',
             error: null,
+            endofTeamsData: false,
         };
     }
 
     componentDidMount() {
-        this.props.actions.getTeams(0, TEAMS_PER_PAGE);
+        this.props.actions.getTeams(0, TEAMS_PER_PAGE, true);
+    }
+
+    fetchMoreTeams = () => {
+        const {actions, totalTeamsCount} = this.props;
+        const BATCH_OF_TEAMS_TO_LOAD = 4; // Load 4 extra teams
+        const NEW_TEAMS_PER_PAGE = this.props.listableTeams.length + BATCH_OF_TEAMS_TO_LOAD;
+        actions.getTeams(0, NEW_TEAMS_PER_PAGE, true);
+        if (this.props.listableTeams.length === totalTeamsCount) {
```

Comment:
Should this be an `await` if we want to deduce the `endofTeamsData` state? because we have to wait for the response here? I still think this will be a problem  because the react lifecycle can be faster when compared to redux `this.props.listableTeams.length`

Can we do this with `componentWIllReceiveProps`?

Reply:
@sudheerDev I have changed this implementation, I moved this implementation to within the `InfiniteScroll` component. I figured we can easily deduce that we have no more data to load using the `pageNumber`, `totalTeamsCount`, `TEAMS_PER_PAGE` and therefore stop the executing the callBack.